### PR TITLE
Applitools Eyes branching and merging support

### PR DESCRIPTION
**Pull request by bcjordan**
This PR implements a first pass at integrating [Applitools branch and merge support](http://support.applitools.com/customer/en/portal/articles/2142886-using-multiple-branches-) with CircleCI.
## TODO

Before merge:
- [x] Create full copy back-up of `default` baselines in case of overwrite, etc.
- [x] Perform an initial copy from `default` to `staging`? 
- [x] Create staging, test, production branches using `rake eyes:create[BRANCHNAME]`

```
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:force_copy[default,default-backup-8-18-2016]
development: 📋  Force copying baselines {:branch=>"default", :base=>"default-backup-8-18-2016"}
java -jar .tmputils/applitools-merge.jar -url https://eyes.applitools.com/api/baselines/copybranch?accesskey=$EYES_ACCESS_KEY -n Code.org -o  -s default -t default-backup-8-18-2016
 200 OK
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:create[test]
development: 👶  Creating branch {:branch=>"test"}
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:create[production]
development: 👶  Creating branch {:branch=>"production"}
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:delete[eyes-branch-merger]
development: Deleting branch {:branch=>"eyes-branch-merger"}
java -jar .tmputils/applitools-merge.jar -url https://eyes.applitools.com/api/baselines/copybranch?accesskey=$EYES_ACCESS_KEY -n Code.org -s eyes-branch-merger -t eyes-branch-merger -d
 200 OK
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:create[eyes-branch-merger]
development: 👶  Creating branch {:branch=>"eyes-branch-merger"}
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:force_merge[default,eyes-branch-merger]
development: 💪  Force merging baselines {:branch=>"default", :base=>"eyes-branch-merger"}
java -jar .tmputils/applitools-merge.jar -url https://eyes.applitools.com/api/baselines/copybranch?accesskey=$EYES_ACCESS_KEY -n Code.org -o -s default -t eyes-branch-merger
 200 OK
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:force_merge[default,staging]
development: 💪  Force merging baselines {:branch=>"default", :base=>"staging"}
java -jar .tmputils/applitools-merge.jar -url https://eyes.applitools.com/api/baselines/copybranch?accesskey=$EYES_ACCESS_KEY -n Code.org -o -s default -t staging
 200 OK
Brians-MacBook-Pro:code-dot-org brian$ rake eyes:force_copy[default,staging]
development: 📋  Force copying baselines {:branch=>"default", :base=>"staging"}
java -jar .tmputils/applitools-merge.jar -url https://eyes.applitools.com/api/baselines/copybranch?accesskey=$EYES_ACCESS_KEY -n Code.org -o  -s default -t staging
 200 OK
```

After merge:
- [ ] Monitor merge behavior

### CODE REVIEW

File: dashboard/test/ui/step_definitions/eyes_steps.rb [hunk 1]
```diff
@@ -35,4 +51,5 @@ def ensure_eyes_available
   # Force eyes to use a consistent host OS identifier for now
   # BrowserStack was reporting Windows 6.0 and 6.1, causing different baselines
   @eyes.host_os = ENV['APPLITOOLS_HOST_OS']
+  @eyes.log_handler = Logger.new(STDOUT)
```
Comment (bcjordan) [inline dashboard/test/ui/step_definitions/eyes_steps.rb]: I believe this is overly verbose, TODO remove.

File: bin/circle/merge-base [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env ruby
+
+require_relative '../../lib/cdo/git_utils'
+
+`git config --global user.email "circleci-github@example.com"`
+`git config --global user.name "CircleCI"`
+
+`git merge origin/#{GitUtils.circle_pr_branch_base_no_origin}`
```
Comment (bcjordan) [inline bin/circle/merge-base]: TODO make this handle non-PR runs (introduce helper that catches & sets to staging / default?)

**Comment by bcjordan** [comment]
@islemaster PTAL 👍  (tests still in progress)

File: .gitignore [hunk 3]
```diff
@@ -3,6 +3,7 @@
 *~
 .DS_Store
 .dropbox
+.tmputils
```
Comment (bcjordan) [inline .gitignore]: TODO: this is causing a warning on CircleCI builds. Add a .tmputils/.gitkeep?

File: dashboard/test/ui/step_definitions/eyes_steps.rb [hunk 4]
```diff
@@ -1,4 +1,8 @@
+$LOAD_PATH.unshift File.expand_path('../../../../../lib', __FILE__)
```
Comment (islemaster) [inline dashboard/test/ui/step_definitions/eyes_steps.rb:1]: 0_o Is there a better way to express this?

**Comment by islemaster** [comment]
Awesome, what I can see LGTM.  Thanks for the overview in person today.

**Pull request merged**